controllers/certificaterequest_controller.go

```diff
@@ -70,6 +71,14 @@ func (r *CertificateRequestReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		return ctrl.Result{}, nil
 	}
 
+	if r.CheckApprovedCondition {
+		// If CertificateRequest has not been approved or is denied, exit early.
+		if !apiutil.CertificateRequestIsApproved(cr) || apiutil.CertificateRequestIsDenied(cr) {
```

Comment:
If the request has been *denied*, should the CertificateRequest be marked as Failed? Here we just ignore it it seems

Reply:
Indeed, this is also what we do in cert-manager-controller. Should we? We don't cover that or say either way I believe in the current design document https://github.com/jetstack/cert-manager/pull/3797/files

We could add the `Ready=Failed`? The only cases I could see this mattering are for clearer user parsing of the state, as well as potentially simplifying clients operating over CRs (they only then need to check for Ready, rather than both Denied and Ready).

I'm not too bothered either way, but semantically a Denied condition should be a terminal state so setting the Ready=False is superfluous.

Reply:
fwiw, the upstream CSR type does *not* have a Ready condition at all. The presence of `status.certificate != nil` or a Denied condition determines if the state is terminal

Reply:
To copy over the discussion from slack, we are going to mark Denied CertificateRequests as Failed.